### DISCUSSION TITLE
Clarification on @wdio/sync mode deprecation and how to migrate

### DISCUSSION THREAD

**Discussion by jamesmortensen** [webdriverio/webdriverio#7267] [Q&A]
@wdio/sync is going away. Thanks goes out to the team and community for attempting to breathe new life into it.  It will be missed, but also it seems we're all ready to move forward.  It's already removed from the CLI wizard. WebdriverIO team says it will be supported until April 2023, when Node 14 reaches it's end of life.

But the question that remains is this:  Will @wdio/sync still be available in WebdriverIO v8, or will those of us still using sync mode be stuck on WebdriverIO v7.x until we update everything? 

Also, how do you recommend migrating over?  One idea we have is to create a separate folder for async mode tests and use a separate configuration file, like wdio.async.conf.js, when running async mode tests and a different configuration file, wdio.sync.conf.js for sync mode tests.  Is there a better way so that we can migrate files in stages?  We don't want to have to have 2 different codebases.

Another idea was to wrap all the sync mode tests with the sync keyword and run the test runner in async mode, if possible.

Thank you for all the work on WebdriverIO
James

**Comment by christian-bromann** [comment]
> But the question that remains is this: Will @wdio/sync still be available in WebdriverIO v8, or will those of us still using sync mode be stuck on WebdriverIO v7.x until we update everything?

No one has stated that sync mode will be removed as part of v8. Only when Node.js v14 support is dropped which is likely to be the v9 release.

> Is there a better way so that we can migrate files in stages?

We are working on a migration guide that shows how to go about this. Simply speaking you can start writing your test cases in async mode together with sync tests:

```js
describe("my test", () => {
  it("a sync test", () => {
    browser.url("....")
    const title = browser.getTitle()
  })

  it("an async test", async () => {
    await browser.url("....")
    const title = await browser.getTitle()
  })
})
```

Both modes work nicely with each other. With that you can migrate test by test until everything is moved over and as a last step you remove `@wdio/sync` from your dependency list.
